lib/core/kotekanTrackers.cpp

```diff
@@ -0,0 +1,68 @@
+#include "kotekanTrackers.hpp"
+
+#include "kotekanLogging.hpp" // for ERROR_NON_OO
+
+#include "fmt.hpp"  // for format, fmt
+#include "json.hpp" // for json, basic_json<>::value_type
+
+#include <functional> // for _Bind_helper<>::type, _Placeholder, bind, _1, placeholders
+#include <stdexcept>  // for runtime_error
+#include <utility>    // for pair
+
+namespace kotekan {
+namespace trackers {
+
+KotekanTrackers::KotekanTrackers() {}
+
+KotekanTrackers::~KotekanTrackers() {
+    restServer::instance().remove_get_callback("/trackers");
+}
+
+KotekanTrackers& KotekanTrackers::instance() {
+    static KotekanTrackers _instance;
+    return _instance;
+}
+
+void KotekanTrackers::register_with_server(restServer* rest_server) {
+    using namespace std::placeholders;
+    rest_server->register_get_callback("/trackers",
+                                       std::bind(&KotekanTrackers::trackers_callback, this, _1));
+}
+
+void KotekanTrackers::trackers_callback(connectionInstance& conn) {
+    nlohmann::json trackers_json = {};
+
+    for (auto& it : trackers) {
+        trackers_json[it.first] = it.second->get_json();
+    }
+
+    conn.send_json_reply(trackers_json);
+}
+
+std::shared_ptr<StatTracker> KotekanTrackers::add_tracker(std::string name, std::string unit,
```

Comment:
I think you should ask for the unique name, and make the map key an `std::pair<std::string, std::string>` with values, `(unique_name, tracker_name)`.  This way you won't run into a problem where you have two stages generating the same key for their stat tracker, and that will make it possible to group stat trackers by stage object names in the returned json.